modules/_labs/select/react/lib/SelectMenu.tsx

```diff
@@ -35,18 +36,28 @@ const fadeOutAnimation = keyframes`
 
 export const menuFadeDuration = 200;
 
-const menuBorderCSS = (error?: ErrorType): CSSObject => {
+const menuBorderCSS = (error?: ErrorType, theme?: EmotionCanvasTheme): CSSObject => {
   let borderColor = inputColors.focusBorder;
   let dividerBorderColor = borderColor;
   let dividerBorderWidth = 1;
-
   if (error === ErrorType.Error) {
-    borderColor = inputColors.error.border;
-    dividerBorderColor = inputColors.error.border;
+    if (theme) {
+      borderColor = theme.canvas.palette.error.main;
+      dividerBorderColor = theme.canvas.palette.error.main;
+    } else {
+      borderColor = inputColors.error.border;
+      dividerBorderColor = inputColors.error.border;
+    }
   } else if (error === ErrorType.Alert) {
-    borderColor = colors.cantaloupe600;
-    dividerBorderColor = inputColors.warning.border;
-    dividerBorderWidth = 2;
+    if (theme) {
+      borderColor = theme.canvas.palette.alert.main;
+      dividerBorderColor = theme.canvas.palette.alert.main;
+      dividerBorderWidth = 2;
+    } else {
+      borderColor = colors.cantaloupe600;
+      dividerBorderColor = inputColors.warning.border;
+      dividerBorderWidth = 2;
+    }
```

Comment:
Same as the code below, theme had to be passed to be able to style different things. Both the methods are different enough however, I would like to simplify these into a cleaner way so I'm open to suggestions

Reply:
Cleaned up with recent changes.